src/Sylius/Bundle/CoreBundle/Doctrine/ORM/ProductRepository.php

```diff
@@ -12,32 +12,26 @@
 namespace Sylius\Bundle\CoreBundle\Doctrine\ORM;
 
 use Doctrine\ORM\QueryBuilder;
-use Pagerfanta\Pagerfanta;
 use Sylius\Bundle\ProductBundle\Doctrine\ORM\ProductRepository as BaseProductRepository;
 use Sylius\Component\Core\Model\ChannelInterface;
-use Sylius\Component\Core\Model\ProductInterface;
 use Sylius\Component\Core\Model\TaxonInterface;
+use Sylius\Component\Core\Repository\ProductRepositoryInterface;
 use Sylius\Component\Product\Model\ArchetypeInterface;
 
 /**
  * @author Paweł Jędrzejewski <pawel@sylius.org>
  * @author Gonzalo Vilaseca <gvilaseca@reiss.co.uk>
  */
-class ProductRepository extends BaseProductRepository
+class ProductRepository extends BaseProductRepository implements ProductRepositoryInterface
 {
     /**
-     * Create paginator for products categorized under given taxon.
-     *
-     * @param TaxonInterface $taxon
-     * @param array          $criteria
-     *
-     * @return Pagerfanta
+     * {@inheritdoc}
      */
     public function createByTaxonPaginator(TaxonInterface $taxon, array $criteria = array())
     {
-        $queryBuilder = $this->getCollectionQueryBuilder();
+        $queryBuilder = $this->createQueryBuilder('o');
```

Comment:
This is not that easy because:

1. Would require to implement a separate class for every resource just to override the alias. Some of them do not need anything custom at all.
2. ``order`` cannot be used for example, so we would have exceptions.
3. Using 'o' as root in every repository is not perfect, but you always know that you can access the root object via this alias and you usually know which repo you are using.
4. Guessing it from the class name is not a good option I believe.